lib/sles4sap_publiccloud_basetest.pm

```diff
@@ -0,0 +1,56 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+# Summary: Base test providing cleanup, post fail and post run hooks for qe-sap-deployment project.
+# https://github.com/SUSE/qe-sap-deployment
+
+package sles4sap_publiccloud_basetest;
+
+use Mojo::Base 'publiccloud::basetest';
+use strict;
+use warnings FATAL => 'all';
+use Exporter 'import';
+use testapi;
+use qesapdeployment;
+
+our @EXPORT = qw(cleanup);
+
+sub cleanup {
+    my ($self) = @_;
+    # Do not run destroy if already executed
+    return if ($self->{cleanup_called});
+    $self->{cleanup_called} = 1;
+    my $inventory_check_cmd = join(" ", ("test", "-f", qesap_get_inventory()));
```

Comment:
Is the idea here to uses the existence of the inventory as sort of internal state?

Reply:
It's a check if ansible destroy does even need to be called. 
If there is no inventory preset, the ansible was most probably not executed. The intention is to avoid "ansible destroy" returing some strage value and causing test failures.
There is probably a better test for that but we can think about it later.

Reply:
I like the idea, no better tests come to my mind too